instrumentation/servlet/servlet-3.0/javaagent-tests/src/test/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/tomcat/TomcatServlet3Test.java

```diff
@@ -0,0 +1,229 @@
+import java.util.stream.IntStream;
+import javax.servlet.Servlet;
+import javax.servlet.ServletException;
+import org.apache.catalina.Context;
+import org.apache.catalina.LifecycleException;
+import org.apache.catalina.core.StandardHost;
+import org.apache.catalina.startup.Tomcat;
+import org.junit.jupiter.api.Assumptions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+public abstract class TomcatServlet3Test extends AbstractServlet3Test<Tomcat, Context> {
+
+  private static final ServerEndpoint ACCESS_LOG_SUCCESS =
+      new ServerEndpoint(
+          "ACCESS_LOG_SUCCESS",
+          "success?access-log=true",
+          SUCCESS.getStatus(),
+          SUCCESS.getBody(),
+          false);
+  private static final ServerEndpoint ACCESS_LOG_ERROR =
+      new ServerEndpoint(
+          "ACCESS_LOG_ERROR",
+          "error-status?access-log=true",
+          ERROR.getStatus(),
+          ERROR.getBody(),
+          false);
+  private final TestAccessLogValve accessLogValue = new TestAccessLogValve();
+
+  @Override
+  protected void configure(HttpServerTestOptions options) {
+    super.configure(options);
+    options.setExpectedException(new ServletException(EXCEPTION.getBody()));
+    options.setContextPath("/tomcat-context");
+    options.setTestError(testError());
+  }
+
+  public boolean testError() {
+    return false;
+  }
+
+  @Override
+  protected SpanDataAssert assertResponseSpan(
+      SpanDataAssert span, SpanData parentSpan, String method, ServerEndpoint endpoint) {
+    if (NOT_FOUND.equals(endpoint)) {
+      span.satisfies(s -> assertThat(s.getName()).matches(".*\\.sendError"))
+          .hasKind(SpanKind.INTERNAL)
+          .hasParent(parentSpan);
+    }
+    return super.assertResponseSpan(span, parentSpan, method, endpoint);
+  }
+
+  @Override
+  protected boolean hasResponseSpan(ServerEndpoint endpoint) {
+    return endpoint == NOT_FOUND || super.hasResponseSpan(endpoint);
+  }
+
+  @Override
+  protected Tomcat setupServer() throws Exception {
+    Tomcat tomcatServer = new Tomcat();
+
+    File baseDir = Files.createTempDirectory("tomcat").toFile();
+    baseDir.deleteOnExit();
+    tomcatServer.setBaseDir(baseDir.getAbsolutePath());
+
+    tomcatServer.setPort(port);
+    tomcatServer.getConnector().setEnableLookups(true); // get localhost instead of 127.0.0.1
+
+    File applicationDir = new File(baseDir, "/webapps/ROOT");
+    if (!applicationDir.exists()) {
+      applicationDir.mkdirs();
+      applicationDir.deleteOnExit();
+    }
+
+    Context servletContext =
+        tomcatServer.addWebapp(getContextPath(), applicationDir.getAbsolutePath());
+    // Speed up startup by disabling jar scanning:
+    servletContext.getJarScanner().setJarScanFilter((jarScanType, jarName) -> false);
+
+    //    setupAuthentication(tomcatServer, servletContext)
```

Comment:
since you already deleted the commented out setupAuthentication method might as well delete this line too